lib/file-source/src/checkpointer.rs

```diff
@@ -24,9 +26,34 @@ enum State {
     V1 { checkpoints: Vec<Checkpoint> },
 }
 
+impl Eq for State {}
+
+impl PartialEq for State {
```

Comment:
We should be able to avoid all this if we change `Vec<Checkpoint>` to `BTreeSet<Checkpoint>` or similar, which shouldn't effect the serialized output.

Reply:
True, the serialized `json` is the same.